snuba/subscriptions/executor_consumer.py

```diff
@@ -290,13 +294,21 @@ def poll(self) -> None:
 
             message, result_future = self.__queue.popleft()
 
-            transformed_message = message.replace(
-                self.__result_encoder.encode(
-                    SubscriptionTaskResult(
-                        result_future.task, result_future.future.result()
+            try:
+                transformed_message = message.replace(
+                    self.__result_encoder.encode(
+                        SubscriptionTaskResult(
+                            result_future.task, result_future.future.result()
+                        )
                     )
                 )
-            )
+            except QueryException as exc:
+                cause = exc.__cause__
+                if isinstance(cause, ClickhouseError):
+                    if cause.code in NON_RETRYABLE_CLICKHOUSE_ERROR_CODES:
+                        logger.exception("Error running subscription query %r", exc)
+                    else:
+                        raise SubscriptionQueryException(exc.message)
```

Comment:
wouldn't you always want to raise this since it causes the executor to fail?

Reply:
@volokluev for `NON_RETRYABLE_CLICKHOUSE_ERROR_CODES` errors i assume we don't want to block subscriptions by retrying and that we rather just move on and wait until we run the next subscription query - if that is what you are asking